src/SolutionOptions.C

```diff
@@ -71,6 +71,14 @@ SolutionOptions::SolutionOptions()
     consistentMMPngDefault_(false),
     useConsolidatedSolverAlg_(false),
     useConsolidatedBcSolverAlg_(false),
+    momentumPerturb_(false),
```

Comment:
I do not think that we need to separate out the perturbation of the stress between momentum and the scalars. These parameters are noted below in eigenvalue*_. I would simply add the new types, e.g., eigenvectorPertrub_ and magnitude.